pynest/nest/tests/test_tsodyks2_connection.py

```diff
@@ -0,0 +1,169 @@
+
+        pre_spikes, weight_by_nest = self.do_the_nest_simulation()
+        weight_reproduced_independently = self.reproduce_weight_drift(
+            pre_spikes,
+            self.synapse_parameters["weight"])
+
+        assert np.all(np.abs(np.array(weight_reproduced_independently) - np.array(weight_by_nest)) < 1E-12)
+
+    def do_the_nest_simulation(self):
+        """
+        This function is where calls to NEST reside.
+        Returns the generated pre- and post spike sequences
+        and the resulting weight established by the tsodyks2 synapse.
+        """
+        nest.set_verbosity('M_WARNING')
+        nest.ResetKernel()
+        nest.SetKernelStatus({'resolution': self.resolution})
+
+        neurons = nest.Create(
+            "parrot_neuron",
+            2,
+            params={})
+        presynaptic_neuron = neurons[0]
+        postsynaptic_neuron = neurons[1]
+
+        presynaptic_generator = nest.Create(
+            "poisson_generator",
+            1,
+            params={"rate": self.presynaptic_firing_rate,
+                    "stop": (self.simulation_duration - self.hardcoded_trains_length)})
+
+        # The detector is to save the randomly generated spike trains.
+        spike_detector = nest.Create("spike_detector")
+
+        nest.Connect(presynaptic_generator, presynaptic_neuron,
+                     syn_spec={"synapse_model": "static_synapse"})
+        nest.Connect(presynaptic_neuron + postsynaptic_neuron, spike_detector,
+                     syn_spec={"synapse_model": "static_synapse"})
+        # The synapse of interest itself
+        wr = nest.Create('weight_recorder', 1)
+        nest.CopyModel("tsodyks2_synapse", "tsodyks2_synapse_rec",
+                       {"weight_recorder": wr})
+        nest.Connect(presynaptic_neuron, postsynaptic_neuron,
+                     syn_spec=self.synapse_parameters)
+
+        nest.Simulate(self.simulation_duration)
+
+        all_spikes = nest.GetStatus(spike_detector, keys='events')[0]
+        pre_spikes = all_spikes['times'][all_spikes['senders'] == presynaptic_neuron.tolist()[0]]
+
+        times = wr.get("events", "times")
+        weights = wr.get("events", "weights")
+
+        return (pre_spikes, weights)
+
+    def reproduce_weight_drift(self, _pre_spikes,
+                               _initial_weight):
+        """
+        Returns the total weight change of the synapse
+        computed outside of NEST.
+        The implementation imitates a step-based simulation: evolving time, we
+        trigger a weight update when the time equals one of the spike moments.
+        """
+
+        # These are defined just for convenience,
+        # STDP is evaluated on exact times nonetheless
+        pre_spikes_forced_to_grid = [int(t / self.resolution) for t in _pre_spikes]
+
+        n_steps = 1 + int(np.ceil(self.simulation_duration / self.resolution))
+        w_log = []
+
+        t_lastspike = 0.
+        A = 1.
```

Comment:
I guess `A` here is defined in the article? It should probably be renamed, or at least commented, to better show what it represents. Same with `R_` below.